evm_loader/program/src/account_storage/backend.rs

```diff
@@ -17,6 +20,22 @@ impl<'a> AccountStorage for ProgramAccountStorage<'a> {
         self.clock.unix_timestamp.into()
     }
 
+    fn block_hash(&self, number: U256) -> H256 {
+        if let Some(account) = self.solana_accounts.get(&recent_blockhashes::ID) {

```

Comment:
You should panic and fail solana transaction if recent_blockhashes account is not in the transaction.
Current behavior can be abused to manipulate solidity contract execution

Reply:
fixed